gensim/models/fasttext.py

```diff
@@ -0,0 +1,363 @@
+            max_vocab_size=None, word_ngrams=1, loss='ns', sample=1e-3, seed=1, workers=3, min_alpha=0.0001,
+            negative=5, cbow_mean=1, hashfxn=hash, iter=5, null_word=0, min_n=3, max_n=6, sorted_vocab=1, bucket=2000000,
+            trim_rule=None, batch_words=MAX_WORDS_IN_BATCH):
+
+        self.load = call_on_class_only
+
+        self.initialize_ngram_vectors()
+
+        # params common between Word2Vec and fastText
+        self.sg = int(sg)
+        self.cum_table = None  # for negative sampling
+        self.vector_size = int(size)
+        self.layer1_size = int(size)
+        if size % 4 != 0:
+            logger.warning("consider setting layer size to a multiple of 4 for greater performance")
+        self.alpha = float(alpha)
+        self.min_alpha_yet_reached = float(alpha)  # To warn user if alpha increases
+        self.window = int(window)
+        self.max_vocab_size = max_vocab_size
+        self.seed = seed
+        self.random = random.RandomState(seed)
+        self.min_count = min_count
+        self.sample = sample
+        self.workers = int(workers)
+        self.min_alpha = float(min_alpha)
+        self.hs = hs
+        self.negative = negative
+        self.cbow_mean = int(cbow_mean)
+        self.hashfxn = hashfxn
+        self.iter = iter
+        self.null_word = null_word
+        self.train_count = 0
+        self.total_train_time = 0
+        self.sorted_vocab = sorted_vocab
+        self.batch_words = batch_words
+        self.model_trimmed_post_training = False
+
+        # fastText specific params
+        self.bucket = bucket
+        self.word_ngrams = word_ngrams
+        self.min_n = min_n
+        self.max_n = max_n
+        if self.word_ngrams <= 1 and self.max_n == 0:
+            self.bucket = 0
+        self.wv.min_n = min_n
+        self.wv.max_n = max_n
+
+        self.wv.ngrams_word = {}
+
+        if sentences is not None:
+            if isinstance(sentences, GeneratorType):
+                raise TypeError("You can't pass a generator as the sentences argument. Try an iterator.")
+            self.build_vocab(sentences, trim_rule=trim_rule)
+            self.train(sentences, total_examples=self.corpus_count, epochs=self.iter,
+                start_alpha=self.alpha, end_alpha=self.min_alpha)
+        else:
+            if trim_rule is not None:
+                logger.warning("The rule, if given, is only used to prune vocabulary during build_vocab() and is not stored as part of the model. ")
+                logger.warning("Model initialized without sentences. trim_rule provided, if any, will be ignored.")
+
+    def initialize_ngram_vectors(self):
+        self.wv = FastTextKeyedVectors()
+
+    def build_vocab(self, sentences, keep_raw_vocab=False, trim_rule=None, progress_per=10000, update=False):
+        self.scan_vocab(sentences, progress_per=progress_per, trim_rule=trim_rule)  # initial survey
```

Comment:
Should definitely be using `super` here.

Reply:
Done